publisher/publisher.go

```diff
@@ -0,0 +1,71 @@
+package publisher
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/screwdriver-cd/sd-cmd/config"
+	"github.com/screwdriver-cd/sd-cmd/screwdriver/api"
+	"github.com/screwdriver-cd/sd-cmd/util"
+)
+
+// Publisher is a type to publish sdapi and sdstore.
+// It receives strings which input by a user.
+// If -f option is valid, yaml file will be loaded to commandSpec struct.
+type Publisher struct {
+	inputCommand map[string]string
+	commandSpec  *util.CommandSpec
+}
+
+// Run is a method to publish sdapi and sdstore.
+func (p *Publisher) Run() error {
+	sdAPI := api.New(config.SDAPIURL, config.SDToken)
+	specPath := p.inputCommand["yamlPath"]
+	specResponse, err := sdAPI.PostCommand(specPath, p.commandSpec)
+	if err != nil {
+		return fmt.Errorf("Post failed:%v", err)
+	}
+
+	// Published successfully
+	// Show version number of command published by sd-cmd
+	println(specResponse.Version)
+
+	return nil
+}
+
+// New is a method to Generate new Publisher.
+// Publisher variable will be returned if input command and yaml file is valid.
+func New(inputCommand []string) (p *Publisher, err error) {
+	p = new(Publisher)
+
+	p.inputCommand, err = parseInputCommand(inputCommand)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to parse command:%v", err)
+	}
+
+	p.commandSpec, err = util.LoadYaml(p.inputCommand["yamlPath"])
+	if err != nil {
+		return nil, fmt.Errorf("Yaml load failed:%v", err)
+	}
+
+	return
+}
+
+func parseInputCommand(inputCommand []string) (map[string]string, error) {
+	fs := flag.NewFlagSet(inputCommand[0], flag.ContinueOnError)
+	var (
+		yamlPath = fs.String("f", "sd-command.yaml", "Path of yaml to publish")
+	)
+
+	subCommand := inputCommand[1]
```

Comment:
`subCommand` is not used, therefore I cannot understand why should you parse `subCommand`? I think it should return an error if you don't use `subCommand` with sd-cmd cli.

Reply:
Yes,  now I do not use subCommand in map.
To be honest, I'd like to improve the architecture that like parsing, do something, then parsing.

parsing here
https://github.com/screwdriver-cd/sd-cmd/blob/master/sd-cmd.go#L66

also parsing validating here
https://github.com/screwdriver-cd/sd-cmd/blob/master/util/util.go#L51

Unfortunatly I also add new parsing program using flag.

What I thinking now is to use urfave.
My idea is parsing and validating in one place with urfave's action. 

Urfave can define subcommand and flags with like a configuration centric manner, and also supports default value and help text, so it looks good for me.

Now I'm not fluent about urfave now, and PR size would be bigger, so I don't use that.
 
So... what I should do now is check the subcommand and return error, but it is just for let programmer grasp this code to do.
(Subcommand have checked already on top level.)

I've fixed with concerning like that. 

If you have question, please tell me.

Reply:
Sorry, I misunderstand your comment. It means the idea not to send all input command to this function if some is no used.

Reply:
I've fixed. Thank you!